apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -338,7 +338,7 @@ def _build_tiledb_schema(
                 [
                     dict(
                         _type="ZstdFilter",
-                        level=tiledb_create_options.string_dim_zstd_level(),
+                        level=tiledb_create_options.int64_dim_zstd_level(),
```

Comment:
the dtype of DataFrame dimensions are not hard-wired to any given primitive type - they could be string, int32, float, etc.  This really should be a more generic name.

Perhaps:  `default_datarame_dim_zstd_level`?

Reply:
In keeping with that idea, maybe the config shoudl jsut have two defaults:
* DataFrame.default_dim_zstd_level
* SparseNDArray.default_dim_zstd_level